conan/internal/runner/ssh.py

```diff
@@ -213,60 +208,190 @@ def copy_working_conanfile_path(self):
+        conan_create_cmd = f'{self.remote_conan} create {raw_args} --format json > {remote_json_output}'
+        script_path = _Path(self.remote_create_dir).joinpath("conan_create").as_posix()
+        script_path = self.remote_conn.create_remote_script(conan_create_cmd, script_path, self.is_remote_windows)
 
-        stdout = channel.makefile("r")
-        stderr = channel.makefile("r")
-        return stdout, stderr
+        if self.remote_conn.run_interactive_command(script_path, self.is_remote_windows, f"Running conan create {raw_args}"):
+            self._update_local_cache(remote_json_output)
+
+        self.client.close()
 
-    def update_local_cache(self, json_result):
-        # ('conan list --graph=create.json --graph-binaries=build --format=json > pkglist.json'
-        _Path = pathlib.PureWindowsPath if self.remote_is_windows else pathlib.PurePath
+    def _update_local_cache(self, json_result: str):
+        _Path = pathlib.PureWindowsPath if self.is_remote_windows else pathlib.PurePath
         pkg_list_json = _Path(self.remote_create_dir).joinpath("pkg_list.json").as_posix()
-        pkg_list_command = f"{self.remote_conan} list --graph={json_result} --graph-binaries=build --format=json > {pkg_list_json}"
+        # List every package (built or cached) because local cache could have been deleted
+        pkg_list_command = f"{self.remote_conan} list --graph={json_result} --format=json > {pkg_list_json}"
         _, stdout, _ = self.client.exec_command(pkg_list_command)
         if stdout.channel.recv_exit_status() != 0:
             raise ConanException("Unable to generate remote package list")
 
         conan_cache_tgz = _Path(self.remote_create_dir).joinpath("cache.tgz").as_posix()
+        self.logger.status("Retrieving remote artifacts into local cache...", fg=Color.BRIGHT_MAGENTA)
+        self.logger.verbose("Remote cache tgz: " + conan_cache_tgz)
         cache_save_command = f"{self.remote_conan} cache save --list {pkg_list_json} --file {conan_cache_tgz}"
         _, stdout, _ = self.client.exec_command(cache_save_command)
         if stdout.channel.recv_exit_status() != 0:
             raise ConanException("Unable to save remote conan cache state")
 
-        sftp = self.client.open_sftp()
         with tempfile.TemporaryDirectory() as tmp:
             local_cache_tgz = os.path.join(tmp, 'cache.tgz')
-            sftp.get(conan_cache_tgz, local_cache_tgz)
-            package_list = self.conan_api.cache.restore(local_cache_tgz)
+            self.remote_conn.get(conan_cache_tgz, local_cache_tgz)
+            self.logger.verbose("Retrieved local cache: " + local_cache_tgz)
+            self.conan_api.cache.restore(local_cache_tgz)
+
+
+class RemoteConnection:
+    def __init__(self, client, logger: RunnerOutput):
+        from paramiko.client import SSHClient
+        self.client: SSHClient = client
+        self.logger = logger
```

Comment:
In general, it is better to call this ``output`` or ``runner_output`` insteda of "logger", as it is more aligned with the rest of the codebase naming.